src/c/admin-external-action.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * window.c.AdminExternalAction component
+ * Makes arbitrary ajax requests and update underlying
+ * data from source endpoint.
+ *
+ * Example:
+ * m.component(c.AdminExternalAction, {
+ *     data: {},
+ *     item: rowFromDatabase
+ * })
+ */
+window.c.AdminExternalAction = ((function(m, h, c, _){
+    return {
+        controller: function(args){
+            let builder = args.data,
+                complete = m.prop(false),
+                error = m.prop(false),
+                fail = m.prop(false),
+                data = {},
+                item = args.item;
+
+            builder.requestOptions.config = (xhr) => {
+                if(window.document.querySelectorAll('meta[name="csrf-token"]').length > 0){
+                    xhr.setRequestHeader('X-CSRF-Token', window.document.querySelectorAll('meta[name="csrf-token"]')[0].content);
+                }
+            };
+
+            const reload = _.compose(builder.model.getRow, h.idVM.id(item[builder.updateKey]).parameters),
+                  l = m.postgrest.loader(builder.requestOptions, m.request);
+
+            const reloadItem = (data) => {
+                reload().then(updateItem);
+            };
+
+            const requestError = (err) => {
+                console.log('err', err);
```

Comment:
Actually it does make sense to log the error. x) So we should only remove the requestOptions log, right?